File: SimpleHabit/app/src/main/java/com/me/simplehabit/data/models/BaseModel.java
package com.me.simplehabit.data.models;

import android.content.Context;

import com.me.simplehabit.network.NetworkDataAgent;
import com.me.simplehabit.network.SimpleHabitDataAgent;
import com.me.simplehabit.persistence.SimpleHabitDatabase;

abstract class BaseModel {

    SimpleHabitDataAgent mDataAgent;
    SimpleHabitDatabase mDatabase;

    BaseModel(Context context){
        mDataAgent= NetworkDataAgent.getInstance();
        mDatabase=SimpleHabitDatabase.getDatabase(context);
    }

}

File: SimpleHabit/app/src/main/java/com/me/simplehabit/data/models/BaseModel.kt
package com.me.simplehabit.data.models

import android.content.Context

import com.me.simplehabit.network.NetworkDataAgent
import com.me.simplehabit.network.SimpleHabitDataAgent
import com.me.simplehabit.persistence.SimpleHabitDatabase

abstract class BaseModel {

    protected val mDataAgent: SimpleHabitDataAgent = NetworkDataAgent
    protected lateinit var mDatabase: SimpleHabitDatabase

    fun initDatabase(context: Context){
       mDatabase = SimpleHabitDatabase.getDatabase(context)
    }

}
